packages/features/pbac/services/permission.service.ts

```diff
@@ -0,0 +1,93 @@
+import type {
+  PermissionString,
+  Resource,
+  CrudAction,
+  CustomAction,
+  PermissionDetails,
+} from "../types/permission-registry";
+import { PERMISSION_REGISTRY } from "../types/permission-registry";
+
+export class PermissionService {
+  validatePermission(permission: PermissionString): boolean {
+    const [resource, action] = permission.split(".") as [Resource, CrudAction | CustomAction];
+    return !!PERMISSION_REGISTRY[resource]?.[action];
+  }
+
+  validatePermissions(permissions: PermissionString[]): boolean {
+    return permissions.every((permission) => this.validatePermission(permission));
+  }
+
+  // Helper function to check if a permission matches a pattern (including wildcards)
+  permissionMatches(pattern: PermissionString, permission: PermissionString): boolean {
+    // Handle full wildcard
+    if (pattern === "*.*") return true;
+
+    const [patternResource, patternAction] = pattern.split(".") as [
+      Resource | "*",
+      CrudAction | CustomAction | "*"
+    ];
+    const [permissionResource, permissionAction] = permission.split(".") as [
+      Resource,
+      CrudAction | CustomAction
+    ];
+
+    // Check if resource matches (either exact match or wildcard)
+    const resourceMatches = patternResource === "*" || patternResource === permissionResource;
+
+    // Check if action matches (either exact match or wildcard)
+    const actionMatches = patternAction === "*" || patternAction === permissionAction;
+
+    return resourceMatches && actionMatches;
+  }
+
+  // Helper function to create a permission string
+  createPermissionString(
+    resource: Resource | "*",
+    action: CrudAction | CustomAction | "*",
+    isCustom = false
+  ): PermissionString {
+    const prefix = isCustom ? "custom:" : "";
+    return `${prefix}${resource}.${action}` as PermissionString;
+  }
```

Comment:
From DB layer to a more presentable usage layer. 

We will pass in strings like "eventType.update" instead of passing in a complex object